src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,40 @@
 package mate.academy;
 
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final CopyOnWriteArrayList<EventListener> listeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executorService = Executors.newCachedThreadPool();
 
+    public void registerListener(EventListener listener) {
+        listeners.addIfAbsent(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : listeners) {
+            executorService.submit(() -> listener.onEvent(event));
+        }
     }
 
     public void shutdown() {
-
+        executorService.shutdown(); // Disable new tasks from being submitted
+        try {
+            if (!executorService.awaitTermination(60, TimeUnit.SECONDS)) {
+                executorService.shutdownNow(); // Cancel currently executing tasks
+                if (!executorService.awaitTermination(60, TimeUnit.SECONDS)) {
+                    System.err.println("Executor service did not terminate");
```

Comment:
Repeatedly awaiting termination within a loop can be simplified. It's better to use a single awaitTermination call with a longer timeout, or handle the loop more elegantly to avoid code duplication.